packages/datadog-plugin-rhea/src/index.js

```diff
@@ -76,29 +95,28 @@ function createWrapReceiverDispatch (tracer, config) {
           ? msgObj.receiver.options.source.address : 'amq.topic'
 
         const childOf = tracer.extract('text_map', msgObj.message.delivery_annotations)
-        return tracer.trace('rhea.receiver.onmessage', {
+        return tracer.trace('amqp.receive', {
           tags: {
+            'component': 'rhea',
             'resource.name': name,
-            'service.name': config.service || `${tracer._service}-amqp`,
+            'service.name': config.service || tracer._service,
             'span.kind': 'consumer',
             'amqp.link.source.address': name,
             'amqp.link.role': 'receiver'
           },
           childOf
         }, (span, done) => {
           msgObj.delivery._dd = { done, span }
-          wrapDeliveryMethod('accept', 'accepted', msgObj.delivery)
-          wrapDeliveryMethod('release', 'released', msgObj.delivery)
-          wrapDeliveryMethod('reject', 'rejected', msgObj.delivery)
-          wrapDeliveryMethod('modified', 'modified', msgObj.delivery)
-          if (!this.get_option('autoaccept', true)) {
-            return dispatch.apply(this, arguments)
-          } else {
+          wrapDeliveryUpdate(msgObj.delivery)
+          let dispatched
+          if (this.get_option('autoaccept', true)) {
             span.setTag('amqp.delivery.state', 'accepted')
-            const d = dispatch.apply(this, arguments)
+            dispatched = dispatch.apply(this, arguments)
```

Comment:
The previous line is not needed since the delivery will be updated.

Reply:
For the call to `done()`, can it be avoided similar to `send` and handled in a single place? (i.e. processing)

Reply:
> The previous line is not needed since the delivery will be updated.

Currently this is skipped if `autoaccept === true`.

The current logic is:

```
if (autoaccept) {
    span is for duration of message handler
} else {
    span is from message dispatch until the first call to update
}
```

I think there's a way to do this exactly the same as with `send`, but we'd need to address error handling. At the moment, error handling is done around the dispatch call (i.e. the `message` event).  If we call `done()` in `processing`, like with senders, then this will always happen in a future tick, meaning the entire message handler will run while the span is active. I think this is the desired effect, so I think we're okay here. I'll have a look at doing this.

Reply:
> Currently this is skipped if autoaccept === true.

From what I can tell [here](https://github.com/amqp/rhea/blob/24d6098d045244866c2659a57eb1c0a16ada4e0d/lib/link.js#L348), it looks like the delivery state will still be updated when autoaccept=true.